backend/app.js

```diff
@@ -43,4 +43,32 @@ app.post('/profile', (req, res) => {
     res.status(200).send("Profile endpoint under construction.");
 });
 
+app.get('/reset-password/:username', async (req, res) => {
```

Comment:
This should be `app.put()` instead of `app.get()`

Setting an ``app.get()`` route here signifies that this route should only be used for **reading** data from the server. Since we're trying to reset a user's password here, we're not trying to read data from the server, instead we want to **update** data from the server. 

For this case, we want to update the user's password to a new value, therefore the correct HTTP methods would be **HTTP PUT** or **HTTP PATCH** which are protocols for updating a resource in the server but there are slight difference between PUT and PATCH.

Reply:
Additionally, we should be using the userId here instead of the username as the dynamic parameter id because based on your prisma schema, the userId is unique and username is not. If two user have the same username, this route would update the password for both users.

```suggestion
app.put('/reset-password/:userId', async (req, res) => {
```

then i would update your prisma.user.update() to update password based on userId

Reply:
oh actually this is a good catch. also _technically_ speaking this is a `PATCH` since it's a partial update 

> PUT is a technique of altering resources when the client transmits data that revamps the whole resource. PATCH is a technique for transforming the resources when the client transmits partial data that will be updated without changing the whole data.